src/transformState.js

```diff
@@ -5,7 +5,29 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const action of actions) {
+    const { type } = action;
+
+    if (type === 'addProperties') {
+      const { extraData } = action;
+
+      Object.assign(state, extraData);
+    } else if (type === 'removeProperties') {
+      const { keysToRemove } = action;
+
+      for (const key of keysToRemove) {
+        if (state.hasOwnProperty(key)) {
+          delete state[key];
+        }
+      }
+    } else if (type === 'clear') {
+      for (const key in state) {
+        if (state.hasOwnProperty(key)) {
+          delete state[key];
+        }
+      }
+    }
+  }
```

Comment:
Use switch statement, also apply changes which were mentioned on the previous reviews about const { type } = action bacause it isn't necessary